### PR TITLE
Use thumbnails

### PR DESCRIPTION
**Pull request by olovy**
## Description
Use thumbnails (128, 256px wide) in search results and on resource page.

### Tickets involved
[LWS-72](https://jira.kb.se/browse/LWS-72)

### Summary of changes
* Use thumbnail images
* Data structure `Image` for passing around image information (can be extended with license information etc.)
* Pass image to use to `ResourceImage` instead of it digging it out from the page data.
* Let `ResourceImage` take multiple images but only display one so far.
* lazy load instance thumbnails

TODO
* Use `ResourceImage` in `SearchCard`?

### CODE REVIEW

File: lxl-web/src/routes/(app)/[[lang=lang]]/(find)/SearchCard.svelte [hunk 1]
```diff
@@ -7,14 +7,16 @@
 	import { page } from '$app/stores';
 	import placeholder from '$lib/assets/img/placeholder.svg';
 	import getTypeIcon from '$lib/utils/getTypeIcon';
+	import type { AuthImageResolution } from '$lib/utils/auxd';
```
Comment (jesperengstrom) [inline lxl-web/src/routes/(app)/[[lang=lang]]/(find)/SearchCard.svelte:10]: Should import from `auxd.types.ts`?

Comment (olovy) [reply lxl-web/src/routes/(app)/[[lang=lang]]/(find)/SearchCard.svelte:10]: Yes!

**Review by jesperengstrom — APPROVED** [review]
LGTM, nice work! ⭐

File: lxl-web/src/lib/utils/auxd.ts [hunk 2]
```diff
@@ -1,51 +1,71 @@
...
-			}
-		});
-		return imageLinks;
-	} else {
-		return [];
+function toImage(imageObject: KbvImageObject, recordId: string): Image {
+	const map1 = (i: KbvImageObject) =>
+		({
+			url: i[JsonLd.ID] || i[Owl.SAME_AS][0][JsonLd.ID],
+			widthṔx: Number.parseInt(i.width || ''),
```
Comment (lrosenstrom) [inline lxl-web/src/lib/utils/auxd.ts:19]: Remove special character in variable name?

File: lxl-web/src/lib/utils/auxd.ts [hunk 3]
```diff
@@ -1,51 +1,71 @@
...
 
-function getInstances(mainEntity) {
-	if (mainEntity['@reverse'] && mainEntity['@reverse']['instanceOf']) {
-		return mainEntity['@reverse']['instanceOf'];
-	} else {
-		return [];
+export function auxdAuth(i: undefined, secret: string): undefined;
+export function auxdAuth(i: Image, secret: string): AuthImage;
+export function auxdAuth(i: ImageResolution, secret: string): AuthImageResolution;
+export function auxdAuth(
```
Comment (lrosenstrom) [inline lxl-web/src/lib/utils/auxd.ts:59]: Maybe come upp with a more descriptive name for this function?

**Review by lrosenstrom — APPROVED** [review]
Nice! Looks good! :ship:

File: lxl-web/src/routes/(app)/[[lang=lang]]/(find)/SearchCard.svelte [hunk 4]
```diff
@@ -7,14 +7,16 @@
...
 	import getTypeIcon from '$lib/utils/getTypeIcon';
+	import type { AuthImageResolution } from '$lib/utils/auxd.types';
 
 	export let item: {
 		'@id': string;
 		'@type': string;
 		'card-heading': ResourceData;
 		'card-body': ResourceData;
-		imageUri: string;
+		image: AuthImageResolution | undefined;
```
[the remark this answers is not in the document]
Comment (olovy) [reply lxl-web/src/routes/(app)/[[lang=lang]]/(find)/SearchCard.svelte]: Yes, nice

File: lxl-web/src/lib/utils/auxd.ts (hunk 3 above)
Comment (olovy) [reply lxl-web/src/lib/utils/auxd.ts]: Trying with the "secure link" terminology from nginx

**Pull request merged**